test_unstructured/partition/xlsx/test_xlsx.py

```diff
@@ -255,3 +263,153 @@ def test_partition_xlsx_with_more_than_1k_cells():
         partition_xlsx("example-docs/more-than-1k-cells.xlsx")
     finally:
         sys.setrecursionlimit(old_recursion_limit)
+
+
+# ------------------------------------------------------------------------------------------------
+# UNIT TESTS
+# ------------------------------------------------------------------------------------------------
+# These test components used by `partition_xlsx()` in isolation such that all edge cases can be
+# exercised.
+# ------------------------------------------------------------------------------------------------
+
+
+class Describe_SubtableParser:
+    """Unit-test suite for `unstructured.partition.xlsx._SubtableParser` objects."""
```

Comment:
Because subtable-parsing is now extracted from the `partition_xlsx()` monolith we are able to thoroughly test edge cases without needing create a separate XLSX file for each case. Also they run extremely quickly, a few milliseconds for each case.